cms/api.py

```diff
@@ -168,6 +169,52 @@ def ensure_program_product_index() -> cms_models.ProgramIndexPage:
     return program_index
 
 
+def ensure_signatory_index() -> cms_models.ProgramIndexPage:
+    """
+    Same as ensure_product_index, but operates on programs instead.
+    """
+    home_page = get_home_page()
+    signatory_index = Page.objects.filter(
+        content_type=ContentType.objects.get_for_model(cms_models.SignatoryIndexPage)
+    ).first()
+    if not signatory_index:
+        signatory_index = cms_models.SignatoryIndexPage(title="Signatories")
+        home_page.add_child(instance=signatory_index)
+        signatory_index.save_revision().publish()
+
+    if signatory_index.get_children_count() != cms_models.SignatoryPage.objects.count():
+        for signatory_page in cms_models.SignatoryPage.objects.all():
+            signatory_page.move(signatory_index, "last-child")
+        log.info("Moved signatory pages under signatory index page")
+    return signatory_index
+
+
+def ensure_certificate_index() -> cms_models.ProgramIndexPage:
+    """
+    Same as ensure_product_index, but operates on programs instead.
+    """
+    home_page = get_home_page()
+    certificate_index = Page.objects.filter(
+        content_type=ContentType.objects.get_for_model(cms_models.CertificateIndexPage)
+    ).first()
+
+    if certificate_index and certificate_index.slug != CERTIFICATE_INDEX_SLUG:
+        certificate_index.slug = CERTIFICATE_INDEX_SLUG
+        certificate_index.save()
+
+    if not certificate_index:
+        cert_index_content_type, _ = ContentType.objects.get_or_create(
+            app_label="cms", model="certificateindexpage"
+        )
+        certificate_index = cms_models.CertificateIndexPage(
+            title="Certificate Index Page",
+            content_type_id=cert_index_content_type.id,
+            slug=CERTIFICATE_INDEX_SLUG,
+        )
+        home_page.add_child(instance=certificate_index)
```

Comment:
Shouldn't we move the existing certificate pages under this index page once created? e.g. something like (from xPRO):

```
    if (
        certificate_index.get_children_count()
        != cms_models.CertificatePage.objects.count()
    ):
        for cert_page in cms_models.CertificatePage.objects.all():
            cert_page.move(certificate_index, "last-child")
        log.info("Moved certificate pages under certificate index page")

```

Reply:
This will not work in xOnline. The certificates are child of courses if we move them in child of certificate index page they got removed from course and then certificate learner view and anonymous view does not work.

Reply:
What is the parent of certifiicates in xPRO? My perception was that they are same for xPRO and MITxOnline.

Reply:
I've tested this in xPro. It is also removing the certificate learner view and anonymous view from xPro as well. This peace of code should be removed from xPro as well

Reply:
The certificate moved from Course child to Certificate Index Page child. And then you cannot edit this page and cannot view the certificate. by moving the certificate again manually from Certificate Index Page to Course Child page everything started working again.

Reply:
Well, this makes sense now Since the ensure_index_pages are only used in `configure_wagtail` that's run with the fresh installation of the app locally. This is the reason we might not have seen any issues though because in a fresh installation there are no course associates certificate pages.

Reply:
But this can be a problematic if the command started by any need in future it'll move all the certificates from courses to Certificate index page.